generated/authorization/microsoft.authorization/2021-07-01-preview/types.md

```diff
@@ -21,7 +21,7 @@
 * **type**: 'Microsoft.Authorization/accessReviewScheduleDefinitions' (ReadOnly, DeployTimeConstant): The resource type
 
 ## Resource Microsoft.Authorization/accessReviewScheduleDefinitions/instances@2021-07-01-preview
-* **Valid Scope(s)**: Subscription
+* **Valid Scope(s)**: Tenant, Subscription
```

Comment:
This is an interesting finding: whether a resource is ReadOnly may depend on the target scope of a deployment. For example, `Microsoft.Authorization/accessReviewScheduleDefinitions/instances@2021-07-01-preview` is PUTable [at subscription scope](https://github.com/Azure/azure-rest-api-specs/blob/74dbcfee45a6634d0043e478aba324796d98940c/specification/authorization/resource-manager/Microsoft.Authorization/preview/2021-07-01-preview/authorization-AccessReviewCalls.json#L331-L437) but read-only [at tenant scope](https://github.com/Azure/azure-rest-api-specs/blob/74dbcfee45a6634d0043e478aba324796d98940c/specification/authorization/resource-manager/Microsoft.Authorization/preview/2021-07-01-preview/authorization-AccessReviewCalls.json#L899-L945). 

I think this is a pretty rare edge case as far as Azure types are concerned, but it does make me wonder if a `Flags` property on resources is the right place to store a `ReadOnly` annotation, since which flags should be applied may be determined by target scope. If we were planning to revisit the `ScopeType` property as part of the extensibility work, combining that and flags into a more general metadata field might be the right thing to do.